pennylane/operation.py

```diff
@@ -3037,7 +3037,9 @@ def convert_to_opmath(op):
         Operator: An operator using the new arithmetic operations, if relevant
     """
     if isinstance(op, qml.Hamiltonian):
-        return qml.dot(*op.terms())
+        c, ops = op.terms()
+        ops = tuple(convert_to_opmath(o) for o in ops)
```

Comment:
Here we probably could have gone either way here, but I just generally have a preference for tuples.

They are more memory and time-efficient since they don't have to pre-allocate space for potential mutations and modifications.  They can also be faster on lookup.  Python is able to do a lot more optimizations since it doesn't need to prioritize the ability to resize.

They are also protected from mutation.  If we want to protect against mutation, they are a good way to go. Especially since they are protected against mutation, we don't have to worry about making copies.  The copy reason is why I want to turn `tape.operations` and `tape.measurements` into tuples eventually.

Reply:
Thanks that is very helpful, didnt know some of that!